src/signature_mahalanobis_knn/sig_mahal_knn.py

```diff
@@ -124,7 +186,69 @@ def compute_auc_given_dists(self, distances_in, distances_out):
         distances_in = np.nan_to_num(distances_in, max_val * 2)
         distances_out = np.nan_to_num(distances_out, max_val * 2)
 
-        return roc_auc_score(
-            [False] * len(distances_in) + [True] * len(distances_out),
-            np.concatenate([distances_in, distances_out]),
+        y_true = [0] * len(distances_in) + [1] * len(distances_out)
+        y_score = np.concatenate([distances_in, distances_out])
+        roc_auc = roc_auc_score(
+            y_true=y_true,
+            y_score=y_score,
+        )
+
+        if plot:
+            plot_roc_curve(
+                y_true=y_true,
+                y_score=y_score,
+                roc_auc=roc_auc,
+                title=title,
+            )
+
+        return roc_auc
+
+    def compute_auc(
```

Comment:
Why do we need a separate compute_auc function given that there is sklearn built-in roc_auc_score?

Reply:
this is just from your code as this already existed. we compute the conformance scores in this function and pass it

happy to merge these two methods together if you'd like

Reply:
I see, I suggest not to put this inside the class of SigMahaKNN as its relationship with the core function of the class (namely, computing outliers score for time-series) is rather weak, and I think there is limited circumstances where a user would like to use this functionality. But this is not blocking.